src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,110 @@
+import duke.parser.WordCommandParser;
+
+/**
+ * Represents a parser that parses user input into a Command object.
+ */
+public class Parser {
+
+    /**
+     * Represents the different types of commands and their corresponding parsers.
+     */
+    public enum Commands {
+
+        // @formatter:off
+        /*
+         * The different types of commands and their corresponding parsers.
+         */
+        LIST("list", new SimpleCommandParser("list")),
+        MARK("mark", new IndexCommandParser("mark")),
+        UNMARK("unmark", new IndexCommandParser("unmark")),
+        DELETE("delete", new IndexCommandParser("delete")),
+        FIND("find", new WordCommandParser("find")),
+        TODO("todo", new TodoCommandParser()),
+        DEADLINE("deadline", new DeadlineCommandParser()),
+        EVENT("event", new EventCommandParser());
+        // @formatter:on
+
+        /**
+         * The string representation of the command.
+         */
+        public final String commandString;
+        private final CommandParser commandParser;
+
+        /**
+         * Constructs a new command with the specified string representation and one
+         * parameter.
+         *
+         * @param commandString the string representation of the command
+         * @param pattern the pattern that the command takes
+         */
+        private Commands(String commandString, CommandParser commandParser) {
+            this.commandString = commandString;
+            this.commandParser = commandParser;
+        }
+
+        /**
+         * Returns the command with the specified string representation.
+         * Idea was taken from https://www.baeldung.com/java-enum-values#locating-values
+         *
+         * @param commandString the string representation of the command
+         * @return the command with the specified string representation
+         */
+        public static Commands valueOfLabel(String commandString) {
+            for (Commands commands : values()) {
+                if (commands.commandString.equals(commandString)) {
+                    return commands;
+                }
+            }
+            return null;
+        }
+
+        /**
+         * Returns the parser for the command.
+         *
+         * @return the parser for the command
+         */
+        public CommandParser getParser() {
+            return this.commandParser;
+        }
+
+    }
+
+    /**
+     * Parses the user input into a Command object.
+     *
+     * @param input the user input
+     * @return the Command object
+     * @throws UnknownCommandException if the command is not recognised
+     * @throws MissingDescriptionException if the description is missing
+     * @throws IncorrectCommandFormatException if the command is in the wrong format
+     * @throws InvalidIndexException if the index provided is invalid
+     * @throws InvalidTimeFormatException if the time provided is invalid
+     */
+    public Command dispatch(String input)
```

Comment:
I like the command class and how it's organised! Perhaps could reconsider renaming this method to parse as I couldn't tell what dispatch meant until I read the javadoc comment

Reply:
I am considering it, but I already have another method called parse. That parse will be overridden for each command depending on how the command is meant to be parsed. Dispatch is used to run that parse.